Tests/DatadogTests/Datadog/RUM/RUMMonitor/Scopes/RUMSessionScopeTests.swift

```diff
@@ -75,91 +75,109 @@ class RUMSessionScopeTests: XCTestCase {
 
```

Comment:
Here, I updated `RUMSessionScope` tests to transitively cover all cases in `RUMOffViewEventsHandlingRule` used internally. I took this approach instead of abstracting `RUMOffViewEventsHandlingRule` through interface and mocking it for tests. I wanted to optimise code readability in `RUMSessionScope`, even if the cost was more effort in fuzzy testing.